### PR TITLE
Optimization of FPinfo and complementIn

### PR DESCRIPTION
**Pull request by WeiqunZhang**
## Summary

Use simplified BoxArray in FPinfo and parallelize FPinfo.  Add parallel version of
BoxList::parallelComplementIn and use it in AmrMesh::MakeNewGrids.

## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] are described in the proposed changes to the AMReX documentation, if appropriate

### CODE REVIEW

**Comment by WeiqunZhang** [comment]
This PR has passed AMReX and incflo regression tests on battra.

File: Src/Base/AMReX_BoxArray.cpp [hunk 1]
```diff
@@ -1275,99 +1275,90 @@ BoxArray::complementIn (BoxList& bl, const Box& bx) const
...
-                            }
-                            bl.swap(newbl);
-                        }
+                }
+            }
+        });
+    } else {
+        AMREX_LOOP_3D(cbx, i, j, k,
+        {
+            amrex::ignore_unused(j,k);
```
Comment (atmyers) [inline Src/Base/AMReX_BoxArray.cpp:1339]: could this be incorporated into the `AMREX_LOOP_3D` macro somehow?

Comment (WeiqunZhang) [reply Src/Base/AMReX_BoxArray.cpp:1339]: We certainly could.

**Review by atmyers — APPROVED** [review]

File: Src/Base/AMReX_BoxArray.cpp (hunk 1 above)
Comment (WeiqunZhang) [reply Src/Base/AMReX_BoxArray.cpp:1339]: I will do this in a new PR.

**Pull request merged**